### ISSUE TITLE
OpenRefine incorrectly saves to Wikidata statements with a value over 2,147,483,647

### ISSUE THREAD

**Issue by Tohaomg** [OpenRefine/OpenRefine#2244]
I used OpenRefine 3.3-beta to add statements with a numeric value to Wikidata (statements with _P2769: budget_ to be exact). For all statements which had a value over 2,147,483,647 (2<sup>31</sup>-1) it saved the statements with the value of exactly 2,147,483,647.

For example:
in OpenRefine while preparing a new statement for _Q4294878: Ministry of Justice of Ukraine_ everything was fine, the correct value of 14,341,937,500 was displayed.

![image](https://user-images.githubusercontent.com/58827708/70744381-3d26b400-1d2a-11ea-85fc-e605de59987a.png)

But after saving it to Wikidata, it turned out it saved an incorrect value.

![image](https://user-images.githubusercontent.com/58827708/70744954-80355700-1d2b-11ea-8e5f-4a03537cc3ee.png)

Here is the edit: https://www.wikidata.org/w/index.php?title=Q4294878&type=revision&diff=1074857426&oldid=1055945359

And it did this not only with this value, but all of them bigger than 2,147,483,647.

I used Opera on Windows 10.

My OpenRefine project, just in case:
[Budget-of-Ukraine-for-year-2020.openrefine.tar.gz](https://github.com/OpenRefine/OpenRefine/files/3957651/Budget-of-Ukraine-for-year-2020.openrefine.tar.gz)

So, I suggest you should use Int64 or bigger one instead.

**Comment by wetneb** [comment]
Hi Anton :)

I am very puzzled by this bug, as I cannot reproduce it on my end: I just made a test edit with the same value, and did not get the behaviour you are describing:
https://www.wikidata.org/w/index.php?title=Q4115189&diff=prev&oldid=1075202542

When I preview the edits from your project, I also see the correct value:
![image](https://user-images.githubusercontent.com/309908/70793934-c1198400-1d9c-11ea-91db-ecce48579f2b.png)

Can you check whether the correct value is displayed in the edit preview on your side? (Just use a facet to isolate this row: the preview should contain only that edit)

In any case, it looks like this could be platform-dependent? That is very curious for a Java stack!

**Comment by Tohaomg** [comment]
Well, yes it shows fine in preview:

![image](https://user-images.githubusercontent.com/58827708/70796240-6c7d0580-1daa-11ea-9ebe-88f0456bb0a2.png)

And repeating the edit ended up in correct value being added: https://www.wikidata.org/w/index.php?title=Q4115189&diff=prev&oldid=1075218927

However I am sure it did not work yesterday. You saw my project and you can see my edits ( https://www.wikidata.org/w/index.php?title=Special:Contributions/Tohaomg&offset=20191212184459&limit=167&target=Tohaomg ) - big values do not math.

Probably I will try to reproduce it with another dataset (year 2017 maybe) and write back here the result.

**Comment by Tohaomg** [comment]
Ok, I started a new project for the year 2017, and here what happened:

I prepared an .xls document with data, which I later exported into the project. The data in .xls is correct (correct value of 8,632,128,100):

![image](https://user-images.githubusercontent.com/58827708/70799563-a7833700-1db2-11ea-8f4d-bc6f19d8dd2f.png)

In OpenRefine in the tab "rows" it is correct again:

![image](https://user-images.githubusercontent.com/58827708/70799619-cda8d700-1db2-11ea-8391-e3f2afe20ba4.png)

But in a tab "preview" the bug finally shows up:

![image](https://user-images.githubusercontent.com/58827708/70799782-427c1100-1db3-11ea-85f1-9f6bee9a2f65.png)

Then, it becomes interesting: I terminated the OpenRefine and launched it again, opened the project again (and did not change it in any way) and went to "preview" tab. It showed the correct value:

![image](https://user-images.githubusercontent.com/58827708/70800100-0d23f300-1db4-11ea-90f5-3f3c6ebd58fe.png)

I exported it all to Wikidata and there were no mistakes.

I started a new project for year 2016, did all the same, but never terminated OpenRefine, i.e. I created the project and exported it to Wikidata immediately.

![image](https://user-images.githubusercontent.com/58827708/70802557-6ee75b80-1dba-11ea-9257-a49479c797de.png)

It saved values over 2,147,483,647 incorrectly:

![image](https://user-images.githubusercontent.com/58827708/70802743-e9b07680-1dba-11ea-8e4b-13b8e883c2e0.png)

I again terminated and then launched OpenRefine (just closing the project and then reopening it does not work) and went to "preview" tab:

![image](https://user-images.githubusercontent.com/58827708/70803166-16b15900-1dbc-11ea-8a32-e23d3b5833ca.png)

See the difference with the previous screenshot.

In conclusion, bug only occurs when a project is created and then exported to Wikidata without restarting OpenRefine.

Here is a sheet for year 2015 which you can use to try it yourself:
[UAbudget2015.xlsx](https://github.com/OpenRefine/OpenRefine/files/3961032/UAbudget2015.xlsx)

**Comment by wetneb** [comment]
Very interesting! Thank you so much for your thorough debugging! I will have a look.

**Comment by Tohaomg** [comment]
I have downloaded the OpenRefine again (assuming that your patch is implemented already), but the bug is still there.

![image](https://user-images.githubusercontent.com/58827708/71753990-1113d400-2e8d-11ea-9daf-b6fbc6a19b51.png)

**Comment by wetneb** [comment]
Hi @Tohaomg, this fix has not been released yet. I am hoping to release a 3.3-rc1 soon. In the meantime you can still run OpenRefine from source: https://github.com/OpenRefine/OpenRefine/wiki/Building-OpenRefine-From-Source
